articles/test/test_views.py

```diff
@@ -0,0 +1,250 @@
+            "image": sample_image(),
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Article.objects.count(), 1)
+
+    def test_create_article_without_body(self) -> None:
+        data = {
+            "title": fake.text(),
+            "description": fake.text(),
+            "image": sample_image(),
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Article.objects.count(), 1)
+
+    def test_create_article_without_authentication(self) -> None:
+        data = {
+            "title": fake.text(),
+            "description": fake.text(),
+            "body": fake.text(),
+            "image": sample_image(),
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(Article.objects.count(), 1)
+
+    def test_create_article_with_invalid_image(self) -> None:
+        data = {
+            "title": fake.text(),
+            "description": fake.text(),
+            "body": fake.text(),
+            "image": "invalid_image",
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Article.objects.count(), 1)
+
+    def test_get_articles(self) -> None:
+        response = self.client.get(reverse("articles"), **self.bearer_token)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)  # type: ignore[attr-defined]
```

Comment:
try not to assert the number of items in the database or if you do, use method described in the above reviews

Reply:
ok